src/WS2812FX/effects/ColorWipeBaseEffect.cpp

```diff
@@ -0,0 +1,147 @@
+        transitionStep = 0;
+        transitionSteps = 10; // Transition over 10 frames
+        inTransition = true;
+        needNewColor = false;
+    }
+    
+    if (inTransition) {
+        // Progress the color transition
+        transitionStep++;
+        
+        // Lerp between current and target color indices
+        if (transitionStep >= transitionSteps) {
+            // Transition complete
+            currentColorIndex = targetColorIndex;
+            inTransition = false;
+        } else {
+            // Interpolate between previous and target
+            uint16_t lerpProgress = map(transitionStep, 0, transitionSteps, 0, 256);
+            currentColorIndex = lerp8by8(previousColorIndex, targetColorIndex, lerpProgress);
+        }
+    }
+}
+
+void ColorWipeBaseEffect::fillWipeColors(WS2812FX* strip, uint16_t fractionalPos16) {
+    auto seg = strip->getSegment();
+    auto runtime = strip->getSegmentRuntime();
+    if (!seg || !runtime) return;
+    
+    // Calculate segment length for boundary checking
+    uint16_t segmentLength = runtime->stop - runtime->start;
+    if (segmentLength == 0) return;
+    
+    // Get colors from palette with current indices
+    CRGB color1 = strip->ColorFromPaletteWithDistribution(
+        *strip->getCurrentPalette(), 
+        currentColorIndex + runtime->baseHue, 
+        seg->targetBrightness, 
+        seg->blendType);
+        
+    CRGB color2 = strip->ColorFromPaletteWithDistribution(
+        *strip->getCurrentPalette(), 
+        previousColorIndex + runtime->baseHue, 
+        seg->targetBrightness, 
+        seg->blendType);
+    
+    // Calculate the wipe bar width (minimum 1 pixel, maximum 3 pixels for smooth transition)
+    uint16_t barWidth = min(max(segmentLength / 8, 1), 3);
+    
+    // Convert segment coordinates to 16-bit fractional format
+    uint16_t segmentStart16 = runtime->start * 16;
+    uint16_t segmentEnd16 = runtime->stop * 16;
+    
+    // Map the wave position to segment coordinates, accounting for bar width
+    uint16_t mappedPos16 = map(fractionalPos16, 0, 65535, segmentStart16, segmentEnd16 - barWidth * 16);
+    
+    // Ensure the position stays within segment bounds
+    mappedPos16 = constrain(mappedPos16, segmentStart16, segmentEnd16 - barWidth * 16);
+    
+    // Get the center pixel position for the wipe boundary
+    uint16_t centerPixel = mappedPos16 / 16; // Convert back to pixel index
+    centerPixel = constrain(centerPixel, runtime->start, runtime->stop - 1);
+    
+    // First, fill the entire strip with color2 as the background
+    fill_solid(&strip->leds[runtime->start], segmentLength, color2);
+    
+    // Then add color1 based on direction
+    if (isMovingUp) {
+        // Moving up: fill color1 from start to center position
+        if (centerPixel > runtime->start) {
+            fill_solid(&strip->leds[runtime->start], centerPixel - runtime->start, color1);
```

Comment:
Hmm, the centerPixel is already mapped to the (16bit) start and end. Why do we substract runtime->start here?